src/components/Details.jsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { setLoading, setRecipe } from '../redux/details/detailsSlice';
+import { setPage, setIsSelected, setSearch } from '../redux/navbar/navbarSlice';
+
+const Details = () => {
+  const dispatch = useDispatch();
+  const { id } = useParams();
+  const { recipes } = useSelector((store) => store.recipes);
+  const { recipe, isLoading } = useSelector((store) => store.details);
+
+  useEffect(() => {
+    dispatch(setRecipe(recipes.find((recipe) => recipe.key === id)));
+    dispatch(setPage(recipe.label));
+    dispatch(setIsSelected(false));
+    dispatch(setSearch(''));
+    dispatch(setLoading(false));
+  }, [dispatch, id, recipes, recipe.label]);
```

Comment:
- [ ] [OPTIONAL AND IMPORTNAT] for some weird reason when I reload your details page, there is an error, this is because the state becomes empty on reload and no proper dispatch on details page is loading the recipe state herby no recipe can be found in the state, it will be good if you can fix this properly
![Screenshot from 2023-04-20 21-01-54](https://user-images.githubusercontent.com/66097371/233477506-91200c96-9d4e-4bb6-bed1-0584490f2b36.png)